### PR TITLE
feat: add event types for search

### LINKED ISSUE #1519: Add Event Types
**Describe the feature you'd like**
Add option to filter events depending upon an event type

**Screenshots**
screenshot-1554278620810

**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1519 

Changes: Add filtering using event types

Screenshots for the change:

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@liveHarshit please review i'm not able to show results in the list view but the api call is happening you might find something

**Comment by liveHarshit** [comment]
Why you are using offline database?

**Comment by liveHarshit** [comment]
What is the API response? `@GET("event-types?sort=name")`. You are not specifying the name.

**Comment by aggarwalpulkit596** [comment]
@liveHarshit api response is the list of all event types

**Comment by aggarwalpulkit596** [comment]
> Why you are using offline database?

@liveHarshit  sorry i didn't get that

**Comment by aggarwalpulkit596** [comment]
@liveHarshit https://open-event-api-dev.herokuapp.com/v1/event-types?sort=name this is response to be precise

**Comment by liveHarshit** [comment]
> > Why you are using offline database?
> 
> @liveHarshit sorry i didn't get that

For Dao file

**Comment by liveHarshit** [comment]
> @liveHarshit https://open-event-api-dev.herokuapp.com/v1/event-types?sort=name this is response to be precise

But on what basis you are shorting events. Replace name with the sorting parameter.

**Comment by aggarwalpulkit596** [comment]
@liveHarshit we dont to fetch the event type every time and i've implemented it similar to eventbrite

**Comment by aggarwalpulkit596** [comment]
> > @liveHarshit https://open-event-api-dev.herokuapp.com/v1/event-types?sort=name this is response to be precise
> 
> But on what basis you are shorting events. Replace name with the sorting parameter.

@liveHarshit in this api call i just want list of all the event types so that user can select from one of them

**Comment by liveHarshit** [comment]
> > > @liveHarshit https://open-event-api-dev.herokuapp.com/v1/event-types?sort=name this is response to be precise
> > 
> > 
> > But on what basis you are shorting events. Replace name with the sorting parameter.
> 
> @liveHarshit in this api call i just want list of all the event types so that user can select from one of them

The user should select event type first, then call API for sorting events accordingly.

File: app/src/main/java/org/fossasia/openevent/general/event/EventService.kt [hunk 1]
```diff
@@ -44,6 +50,24 @@ class EventService(
...
 
+    fun getEventTypes(): Flowable<List<EventType>> {
+        val eventsLocationFlowable = eventTypesDao.getAllEventTypes()
+        return eventsLocationFlowable.switchMap {
+            if (it.isNotEmpty())
+                eventsLocationFlowable
+            else
+                eventTypesApi.getEventTypes()
+                    .map {
+                        eventTypesDao.insertEventTypes(it)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventService.kt]: Don't insert these into DB, let them be fetched from network every time

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/event/EventService.kt]: @iamareebjamal okay will revert this

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal @liveHarshit review

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal i'm having trouble understing the query part for loading event can you help me out there rest all is implemented just i need to add event param to search query

**Comment by iamareebjamal** [comment]
I don't understand?

**Comment by aggarwalpulkit596** [comment]
@liveHarshit @iamareebjamal please review

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal any update ?

File: app/src/main/java/org/fossasia/openevent/general/event/EventService.kt [hunk 2]
```diff
@@ -44,6 +48,10 @@ class EventService(
         return eventTopicsDao.getAllEventTopics()
     }
 
+    fun getEventTypes(): Single<List<EventType>> {
+                return eventTypesApi.getEventTypes()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventService.kt]: Indentation

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 3]
```diff
@@ -73,6 +74,9 @@ class SearchFragment : Fragment() {
                 rootView.timeTextView.text = time
             }
         }
+        val type = safeArgs?.stringSavedType
+        if (type.isNullOrBlank()) rootView.eventTypeTextView.text = getString(R.string.anything)
+        else rootView.eventTypeTextView.text = type
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: rootView.eventTypeTextView.text = if (type.isNullOrBlank()) getString(R.string.anything) else type

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        rootView = inflater.inflate(R.layout.fragment_search_type, container, false)
+        val thisActivity = activity
+        if (thisActivity is AppCompatActivity) {
+            thisActivity.supportActionBar?.title = ""
+            thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchTypeFragment.kt]: Use Utils

File: app/src/main/java/org/fossasia/openevent/general/search/SearchTypeViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import io.reactivex.schedulers.Schedulers
+import org.fossasia.openevent.general.event.EventService
+import org.fossasia.openevent.general.event.types.EventType
+import timber.log.Timber
+
+class SearchTypeViewModel(
+    private val eventService: EventService
+) : ViewModel() {
+
+    private val compositeDisposable = CompositeDisposable()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchTypeViewModel.kt]: You're not disposing it

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchTypeViewModel.kt]: @iamareebjamal how to dispose it a little bit of help there

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchTypeViewModel.kt]: See other ViewModels

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchTypeViewModel.kt]: already done please review

**Pull request merged**